File: Bookapp/Bookapp.Web/Modules/ServerTypes/BookShop/BookColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BookRow } from "./BookRow";

export interface BookColumns {
    Id: Column<BookRow>;
    Title: Column<BookRow>;
    Description: Column<BookRow>;
    PageCount: Column<BookRow>;
    PublishDate: Column<BookRow>;
    Excerpt: Column<BookRow>;
}

export class BookColumns extends ColumnsBase<BookRow> {
    static readonly columnsKey = 'BookShop.Book';
    static readonly Fields = fieldsProxy<BookColumns>();
}

File: Bookapp/Bookapp.Web/Modules/BookShop/Book/BookColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Bookapp.BookShop.Columns;

[ColumnsScript("BookShop.Book")]
[BasedOnRow(typeof(BookRow), CheckNames = true)]
public class BookColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    public int PageCount { get; set; }
    public DateTime PublishDate { get; set; }
    public string Excerpt { get; set; }
}